confluent_kafka/src/Consumer.c

```diff
@@ -1386,7 +1386,11 @@ static int Consumer_init (PyObject *selfobj, PyObject *args, PyObject *kwargs) {
 
         rd_kafka_poll_set_consumer(self->rk);
 
-        self->u.Consumer.rkqu = rd_kafka_queue_get_consumer(self->rk);
+        if (!(self->u.Consumer.rkqu = rd_kafka_queue_get_consumer(self->rk))) {
```

Comment:
This is checking a side-effect of not setting group.id, that might change in librdkafka.
Explicitly check that group.id was configured.

You can leave this check in, but make sure that memory resources are freed up in the error case, this is leaking memory.

Reply:
Good call, I didn't think about how closely this couples the validation to the librdkafka implementation. I'll remove it in favor of a check for 'group.id`'s presence. And clean up after myself of course.